### ISSUE TITLE
Generic sensitivity estimation

### ISSUE THREAD

**Issue by bkhelifi** [gammapy/gammapy#4576]
**Is your feature request related to a problem? Please describe.**
The current sensitivity estimator is designed such than only one data set at a fixed alt/az position can be used and the latter is built automatically from the IRFs. 

**Describe the solution you'd like**
It would be nice to add the following features:
- a tool that is usable for non-pointing instrument (e.g. SWGO), ie with a sky position (inclusion of the transit, the variation of declinaison)
- a tool that can use simultaneously several datasets, e.g. SWGO+CTA or EventType1+EventType2+EventType3

I am wondering also if one could generalise by using the Li&Ma stats (which is the TeV standard, the official standard) or the Cash stats (3D analysis)

**Comment by HealthyPear** [comment]
Hi @bkhelifi ,

regarding the first point, me and a collaborator from SWGO are trying to get a sensitivity curve (as soon as he activates an account here I'll tag him).

We immediately got stuck because of a series of cascading problems originating from these issues:
- in HAWC/SWGO one usually organizes the entire analysis in bins of e.g. Core/Energy/Zenith, so what we have now is a code that generates a dataset for each of them and we get a SpectrumDatasetOnOff for each of them. Later they need to be stacked together in order to use the SensitivityEstimator
- HAWC/SWGO IRFs and other objects at this level of the analysis are usually RA/Dec sky maps in bins of energy, which in the case of gammapy translates into `MapDataset`s

During the first naïve reverse engineering experiments, the stacking of the `dataset_on_off` instances failed immediately because the code was asking for the exposures, and when using e.g. `gammapy.makers.utils.make_map_exposure_true_energy` the next issue is that everything in gammapy at this level requires *tables* (like `gammapy.irf.EffectiveAreaTable2D`) and not maps.

The third problem encountered is that for HAWC-like experiments there is no definition of *pointing* and thus no immediate definition of *offset* when referring to the format of these tables.

We recently managed to get an effective area in this format (in particular GADF v0.3) by defining the offset as the zenith distance between the Zenith and the instant position of the transiting source (so like CTA will provide IRFs at different zenith angles because it is effectively pointing there, we would do the same but from Zenith).

(If you think this use case is specific enough, we could open a dedicated issue instead of cramming all the discussion here, since you mention different things in your issue description.)

**Comment by registerrier** [comment]
It seems to me that the `SensitivityEstimator` we have at the moment, is not the right tool in this context.
- If the data are not in the form of a `SpectrumDatasetOnOff`, it might not be worth trying to squeeze them into one.
- When dealing with 'joint' sensitivity, I think that count statistics are no longer relevant. You cannot determine the sensitivity in terms of counts and convert it to a flux, you have to determine it directly as a flux.
All in all a method based on MC simulations might be the more natural solution, even if computationally more intensive.
I think that is the solution that is being explored to get the sensitivity curve of joint analyses combining event types in CTA.

**Comment by HealthyPear** [comment]
> It seems to me that the `SensitivityEstimator` we have at the moment, is not the right tool in this context.
> 
> * If the data are not in the form of a `SpectrumDatasetOnOff`, it might not be worth trying to squeeze them into one.

The fact is that if we didn't have multiple (Map)Datasets, what is currently in gammapy should works for us. It's when we have to stack multiple of them that we get the problems. Right now we managed to get an `EffectiveAreaTable2D` with the caveats above, so we can try to make it work (I think we'll need to do the same with energy dispersion and PSF though).

For reference, the summarized sequence of operation we are trying is this one:
- we start from HAWC-like IRF files (so maps as I explain above) and an events file
- ``maker = MapDatasetMaker(selection=["exposure", "psf", "edisp"])```
- we build a list of datasets using `.get_observations` from a `DataStore`
- we define a pointing in the center of the FoV and a `CircleSkyRegion`
- we produce a `dataset_on_off` using `SpectrumDatasetOnOff.from_spectrum_dataset` from `dataset.to_spectrum_dataset(on_region)`
- we the *should* stack the `dataset_on_off` datasets together, but this doesn't work for the caveats I described above
- we produce a sensitivity table with `SensitivityEstimator` (this works with any single dataset, but of course it's not correct)

**Comment by bkhelifi** [comment]
I agree, Régis... This feature request will imply the creation of a new tool (e.g. an estimator).

Otherwise, I am not sure about MC, maybe a dichotomic search is more adapted.

**Comment by AtreyeeS** [comment]
> All in all a method based on MC simulations might be the more natural solution, even if computationally more intensive.
I think that is the solution that is being explored to get the sensitivity curve of joint analyses combining event types in CTA.

@TarekHC and Juan already have a prototype for using MC simulations to get the sensitivity for a joint analysis (specifically for event types, but can be generalised). We are testing it now, and it can be added in gammapy once its ready

**Comment by QRemy** [comment]
Implemented in  #5637, #5639, #5679
